### PR TITLE
D8 un 289

### PR DESCRIPTION
**Pull request by soda736**
Adding active facets to the breadcrumb trail for search pages.

### CODE REVIEW

File: web/sites/uregni/modules/custom/uregni_breadcrumbs/uregni_breadcrumbs.module [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    $about_us_link[] = Link::fromTextandUrl(t('About us'), Url::fromUri('entity:node/53'));
+    array_splice($links, 1, 0, $about_us_link);
+  }
+
+  // Loop through the links for links that have facet queries added to them and unset the link of the last breadcrumb.
+  // The facets module breadcrumb links them all by default.
+  foreach ($links as $link) {
+    // Strip the - search results text from the page title breadcrumb link.
+    if (preg_match('/([A-Z|a-z]+) - search results/', $link->getText(), $matches)) {
+      $link->setText($matches[1]);
```
Comment (DuttonMa) [inline web/sites/uregni/modules/custom/uregni_breadcrumbs/uregni_breadcrumbs.module:33]: Nice use of regex !

**Review by DuttonMa — CHANGES_REQUESTED** [review]
Looks like this code will work nicely - although if breadcrumbs start to get too complicated it can be a good idea to implement some classes that implement BreadcrumbBuilderInterface to generate breadcrumbs from scratch in certain cases (see NIDirect).

File: web/sites/uregni/modules/custom/uregni_breadcrumbs/uregni_breadcrumbs.module [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * Altering the search view page breadcrumbs, when facets are selected the breadcrumb trail is overwritten.
+ */
+function uregni_breadcrumbs_system_breadcrumb_alter(Breadcrumb &$breadcrumb, RouteMatchInterface $route_match, array $context) {
+  $links = $breadcrumb->getLinks();
+  $route_name = $route_match->getRouteName();
+
+  // The publications view page has an extra breadcrumb.
+  if ($route_name == 'view.publications_search.publication_search_page') {
+    $about_us_link[] = Link::fromTextandUrl(t('About us'), Url::fromUri('entity:node/53'));
```
[the remark this answers is not in the document]
Comment (soda736) [reply web/sites/uregni/modules/custom/uregni_breadcrumbs/uregni_breadcrumbs.module:24]: I've updated the comment to explain why we need the hardcoded link to the about us page.

[the remark this answers is not in the document]
Comment (soda736) [reply web/sites/uregni/modules/custom/uregni_breadcrumbs/uregni_breadcrumbs.module:24]: I've also removed the PublicationViewPage breadcrumb and amended the ViewPageBreadcrumb to add the extra link needed for the publication view page.

**Review by DuttonMa — APPROVED** [review]
Looks good

**Pull request merged**